feat: Add logger to preparation

splasky opened the pull request:
Logger is helpful for generate debug message or info. It can also
disable all debug message when DEBUG flag not set.

jserv commented:
It is redundant to denote "Add logger to log message." Logger is always used for logging. Instead, you have to explain how the logger works along with potential integration efforts.

jserv commented:
Add documentation!

splasky commented:
related #34

jserv commented:
Fix conflicts.

jserv reviewed (changes requested):
"preparation" should be emphasized since no code snippet is using logger so far.

jserv commented on utils/logger.c:
```diff
@@ -0,0 +1,6 @@
+#include "logger.h"
+static int TA_LOG_LEVEL = LV_INFO;
```
You should implement non-trivial logger, which takes circular buffer into consideration.

jserv commented on utils/logger.h:
```diff
@@ -0,0 +1,42 @@
+
+#ifndef UTIL_LOGGER_H
+#define UTIL_LOGGER_H
+
```
Add comments about these level.

Commit:
feat: [WIP]Add logger

Logger is helpful for generate debug message or info. It can also
disable all debug message when DEBUG flag not set.

splasky commented:
Will use the tangle-accelerator's logger as default logger.
